common/src/main/java/com/wynntils/features/inventory/CustomBankPagesFeature.java

```diff
@@ -246,35 +247,47 @@ private void clickPreviousPage() {
     @Override
     protected void onConfigUpdate(ConfigHolder configHolder) {
         String valueString = (String) configHolder.getValue();
+        String fieldName = configHolder.getFieldName();
 
-        List<Integer> originalValues = parseStringToDestinations(valueString);
+        SearchableContainerType containerType = fieldName.equals("bankDestinations") ? SearchableContainerType.BANK : SearchableContainerType.BOOKSHELF;
+
+        List<Integer> originalValues = parseStringToDestinations(valueString, containerType);
 
         if (originalValues == null) {
             configHolder.setValue(configHolder.getDefaultValue());
             return;
         }
 
-        int maxValue =
-                configHolder.getFieldName().equals("bankDestinations") ? MAX_BANK_PAGES : MAX_HOUSING_CONTAINER_PAGES;
+        int maxValue = fieldName.equals("bankDestinations") ? MAX_BANK_PAGES : MAX_HOUSING_CONTAINER_PAGES;
 
         List<Integer> newValues = modifyJumpValues(originalValues, maxValue);
 
-        if (!newValues.equals(originalValues)) {
-            String formattedConfig = newValues.stream().map(Object::toString).collect(Collectors.joining(","));
+        String formattedConfig = newValues.stream().limit(MAX_DESTINATIONS).map(Object::toString).collect(Collectors.joining(","));
 
+        if (!formattedConfig.equals(valueString)) {
             configHolder.setValue(formattedConfig);
         }
     }
 
-    private List<Integer> parseStringToDestinations(String destinationsStr) {
+    private List<Integer> parseStringToDestinations(String destinationsStr, SearchableContainerType containerType) {
         String[] destinationStrings = destinationsStr.split(",");
 
-        if (destinationStrings.length != 6) {
-            return null;
-        }
-
         try {
-            return Arrays.stream(destinationStrings).map(Integer::parseInt).collect(Collectors.toList());
+            List<Integer> destinations = Arrays.stream(destinationStrings)
+                    .limit(MAX_DESTINATIONS)
+                    .map(Integer::parseInt)
+                    .collect(Collectors.toCollection(ArrayList::new));
+
+            if (destinations.size() != MAX_DESTINATIONS) {
```

Comment:
```suggestion
            if (destinations.size() < MAX_DESTINATIONS) {
```

Logically equivalent but clearer.